eng/common/scripts/TypeSpec-Project-Sync.ps1

```diff
@@ -13,115 +15,133 @@ Install-ModuleIfNotInstalled "powershell-yaml" "0.4.1" | Import-Module
+          $result = "https://github.com/$repo.git"
+          break
         }
+        elseif ($remote -match "(.*)?git@github.com:\S+") {
+          $result = "git@github.com:$repo.git"
+          break
+        }
+        else {
+          throw "Unknown git remote format found: $remote"
+        }
+      }
     }
-    finally {
-        Pop-Location
-    }
-
-    return $result
+  }
+  finally {
+    Pop-Location
+  }
+  Write-Host "Found git remote $result"
+  return $result
 }
 
 function InitializeSparseGitClone([string]$repo) {
-    git clone --no-checkout --filter=tree:0 $repo .
-    if ($LASTEXITCODE) { exit $LASTEXITCODE }
-    git sparse-checkout init
-    if ($LASTEXITCODE) { exit $LASTEXITCODE }
-    Remove-Item $sparseCheckoutFile -Force
+  git clone --no-checkout --filter=tree:0 $repo .
+  if ($LASTEXITCODE) { exit $LASTEXITCODE }
+  git sparse-checkout init
+  if ($LASTEXITCODE) { exit $LASTEXITCODE }
+  Remove-Item $sparseCheckoutFile -Force
 }
 
 function GetSpecCloneDir([string]$projectName) {
-    Push-Location $ProjectDirectory
-    try {
-        $root = git rev-parse --show-toplevel
-    }
-    finally {
-        Pop-Location
-    }
-
-    $sparseSpecCloneDir = "$root/../sparse-spec/$projectName"
-    New-Item $sparseSpecCloneDir -Type Directory -Force | Out-Null
-    $createResult = Resolve-Path $sparseSpecCloneDir
-    return $createResult
+  Push-Location $ProjectDirectory
+  try {
+    $root = git rev-parse --show-toplevel
+  }
+  finally {
+    Pop-Location
+  }
+
+  $sparseSpecCloneDir = "$root/../sparse-spec/$projectName"
+  New-Item $sparseSpecCloneDir -Type Directory -Force | Out-Null
+  $createResult = Resolve-Path $sparseSpecCloneDir
+  return $createResult
 }
 
 $typespecConfigurationFile = Resolve-Path "$ProjectDirectory/tsp-location.yaml"
 Write-Host "Reading configuration from $typespecConfigurationFile"
 $configuration = Get-Content -Path $typespecConfigurationFile -Raw | ConvertFrom-Yaml
 
-$pieces = $typespecConfigurationFile.Path.Replace("\","/").Split("/")
+$pieces = $typespecConfigurationFile.Path.Replace("\", "/").Split("/")
 $projectName = $pieces[$pieces.Count - 2]
 
 $specSubDirectory = $configuration["directory"]
 
-if ( $configuration["repo"] -and $configuration["commit"]) {
-    $specCloneDir = GetSpecCloneDir $projectName
-    $gitRemoteValue = GetGitRemoteValue $configuration["repo"]
-
-    Write-Host "Setting up sparse clone for $projectName at $specCloneDir"
-
-    Push-Location $specCloneDir.Path
-    try {
-        if (!(Test-Path ".git")) {
-            InitializeSparseGitClone $gitRemoteValue
-        }
-        UpdateSparseCheckoutFile $specSubDirectory $configuration["additionalDirectories"]
-        git checkout $configuration["commit"]
-        if ($LASTEXITCODE) { exit $LASTEXITCODE }
-    }
-    finally {
-        Pop-Location
+# use local spec repo if provided
+if ($LocalSpecRepoPath) {
+  $specCloneDir = $LocalSpecRepoPath
```

Comment:
Do we also need to checkout the correct branch or at least validate that it matches?

Reply:
I would leave it to user to ensure the path passed in has the right type specs.